### PR TITLE
Generate E2E typescript test report for ReportPortal

### LINKED ISSUE #22543: Generate E2E typescript test report for ReportPortal
### Is your task related to a problem? Please describe

Make it possible to generate test report for ReportPortal https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com

### Describe the solution you'd like

Use [@reportportal/agent-js-mocha](https://github.com/reportportal/agent-js-mocha#readme) library.

[Reporter option](https://github.com/reportportal/agent-js-mocha#reportportalagent-js-mocha):
```
const mochaMain = new Mocha({
  reporter: '@reportportal/agent-js-mocha',
  reporterOptions: {
    endpoint: 'https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/api/v1',
    token: 'RP_TOKEN',
    launch: 'RP_LAUNCH_NAME',
    project: 'RP_PROJECT_NAME',
    description: 'RP_LAUNCH_DESCRIPTION'
  },
  timeout: 250000
});
```
There are dedicated projects:
- project for **devspaces** test reports: https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/ui/#devspaces/dashboard
- project for **che** test reports: https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/ui/#che/dashboard

### Describe alternatives you've considered

_No response_

### Additional context

Downstream issue: https://issues.redhat.com/browse/CRW-1433

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
resolves #22543 

### Screenshot/screencast of this PR

https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/ui/#devspaces/launches/all

### What issues does this PR fix or reference?

resolves #22543 

### How to test this PR?

Use `REPORTER_CONSTANTS` to set reporter parameters 
Then run tests, for example using `tests/e2e/configs/sh-scripts/regressionTestRun.sh`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@nallikaea : could you please share an example of ReportPortal constants to run Dev Spaces functional test suite and send test results to ReportPortal ?:
```
docker run --shm-size=4096m -p 5920:5920 \
 -e TS_SELENIUM_LOAD_PAGE_TIMEOUT=420000 \
 -e TS_SELENIUM_BASE_URL=https://devspaces.apps.ocp413.crw-qe.com \
 -e DELETE_WORKSPACE_ON_FAILED_TEST=true \
 -e TS_SELENIUM_START_WORKSPACE_TIMEOUT=120000 \
 -e NODE_TLS_REJECT_UNAUTHORIZED=0 \
 -e VIDEO_RECORDING=true \
 -e E2E_OCP_CLUSTER_API_URL=https://api.ocp413.crw-qe.com:6443 \
 -e TS_SELENIUM_LOG_LEVEL=TRACE \
 -e TS_WAIT_LOADER_PRESENCE_TIMEOUT=120000 \
 -e TS_COMMON_DASHBOARD_WAIT_TIMEOUT=30000 \
 -e OCP_INFRA=PSI \
 -v /tmp:/usr/local/bin:Z \
 -v /home/hudson/.kube:/home/seluser/.kube:Z \
 -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e:/tmp/e2e:Z \
 -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e/video:/tmp/ffmpeg_report:Z \
 -e TS_SELENIUM_VALUE_OPENSHIFT_OAUTH=true \
 -e TS_OCP_LOGIN_PAGE_PROVIDER_TITLE=htpasswd \
 -e TS_SELENIUM_OCP_USERNAME=******* \
 -e 'TS_SELENIUM_OCP_PASSWORD=******' \
 -e TS_COMMON_DASHBOARD_WAIT_TIMEOUT=120_000 \
 -e TEST_SUITE=functional-test-suite \
 -e OCP_VERSION=4.13 \
 -e ARCH_VERSION=x86_64 quay.io/eclipse/che-e2e:next
```

**Comment by nallikaea** [comment]
docker run --shm-size=4096m -p 5920:5920 \
 -e TS_SELENIUM_BASE_URL=https://devspaces.apps.ocp413.crw-qe.com \
 -e TS_SELENIUM_LOG_LEVEL=TRACE \
 -v /tmp:/usr/local/bin:Z \
 -v /home/hudson/.kube:/home/seluser/.kube:Z \
 -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e:/tmp/e2e:Z \
 -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e/video:/tmp/ffmpeg_report:Z \
 -e TS_SELENIUM_OCP_USERNAME=******* \
 -e 'TS_SELENIUM_OCP_PASSWORD=******' \
 -e TEST_SUITE=functional-test-suite \
 -e OCP_VERSION=4.13 \
 -e SAVE_RP_REPORT_DATA=true \
 -e RP_API_KEY=<USER_APP_KEY_OR_TOKEN> \
 -e RP_IS_LOCAL_SERVER=false \
 -e RP_USE_PERSONAL=false \

 quay.io/eclipse/che-e2e:next

**Comment by dmytro-ndp** [comment]
@nallikaea : thank you

For the sake of simplicity, I would use constant `OCP_INFRA=["PSI", "AWS", "IBM Z", "IBM Power"]` instead of `ARCH_VERSION`.

The point is that when it turns to CRW Jenkins CI, `typescript-tests` pipeline is running tests using x86_64 Jenkins nodes doesn't matter if the testing OCP arch is _amd64_, or _s390x_.
The only difference is value of `OCP_INFRA` parameter of `typescript-tests` pipeline.

**Comment by nallikaea** [comment]
> @nallikaea : thank you
> 
> For the sake of simplicity, I would use constant `OCP_INFRA=["PSI", "AWS", "IBM Z", "IBM Power"]` instead of `ARCH_VERSION`.
> 
> The point is that when it turns to CRW Jenkins CI, `typescript-tests` pipeline is running tests using x86_64 Jenkins nodes doesn't matter if the testing OCP arch is _amd64_, or _s390x_. The only difference is value of `OCP_INFRA` parameter of `typescript-tests` pipeline.

okay, replaced in new commit

File: tests/e2e/constants/REPORTER_CONSTANTS.ts [hunk 1]
```diff
@@ -44,10 +58,89 @@ export const REPORTER_CONSTANTS: {
...
+
+	/**
+	 * reportPortal app key or user token
+	 */
+	RP_API_KEY: process.env.RP_API_KEY || '',
+
+	/**
+	 * user name on ReportPortal
+	 */
+	RP_USER: process.env.RP_USER || process.env.USER || 'superadmin',
```
Comment (dmytro-ndp) [inline tests/e2e/constants/REPORTER_CONSTANTS.ts]: From security perspective I would prefer empty RP user name by default.

**Comment by dmytro-ndp** [comment]
@nallikaea : thanks for the fixup.

I tried to run tests with parameters you shared [1] and using personal RP access token [2], but had got an error:
```
01:10:18  Error: Request failed with status code 403: {"errorCode":4003,"message":"You do not have enough permissions."}
01:10:18  URL: https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/api/v1/superadmin_personal/launch
01:10:18  method: POST
01:10:18      at /tmp/e2e/node_modules/@reportportal/client-javascript/lib/rest.js:41:15
01:10:18      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

It seemed to be missing ` -e RP_USER` in the test run command.

[1] https://github.com/eclipse/che/pull/22590#issuecomment-1775372967
[2] https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/13528/consoleFull

**Comment by dmytro-ndp** [comment]
Second try with RT_USER parameter and single testcase was successful: https://reportportal-crw.apps.ocp-c1.prod.psi.redhat.com/ui/#devspaces/launches/all/132 , https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/13529/console

At the same time I couldn't find info about test OCP version = 4.14 and OCP infra = PSI in RP test launch.
Test run command:
```
docker run --shm-size=4096m -p 5920:5920 -e TS_SELENIUM_LOAD_PAGE_TIMEOUT=420000 -e TS_SELENIUM_BASE_URL=https://devspaces.apps.ocp414.crw-qe.com/ -e DELETE_WORKSPACE_ON_FAILED_TEST=true -e TS_SELENIUM_START_WORKSPACE_TIMEOUT=120000 -e NODE_TLS_REJECT_UNAUTHORIZED=0 -e VIDEO_RECORDING=true -e E2E_OCP_CLUSTER_API_URL=https://api.ocp414.crw-qe.com:6443/ -e TS_SELENIUM_LOG_LEVEL=TRACE -e TS_WAIT_LOADER_PRESENCE_TIMEOUT=120000 -e TS_COMMON_DASHBOARD_WAIT_TIMEOUT=30000 -e OCP_INFRA=PSI -e OCP_VERSION=4.14 -v /tmp:/usr/local/bin:Z -v /home/hudson/.kube:/home/seluser/.kube:Z -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e:/tmp/e2e:Z -v /mnt/hudson_workspace/workspace/Testing/e2e/basic/typescript-tests/test-repo/tests/e2e/video:/tmp/ffmpeg_report:Z -e USERSTORY=SmokeTest -e TS_SELENIUM_OCP_USERNAME=***** -e 'TS_SELENIUM_OCP_PASSWORD=*****' -e SAVE_RP_REPORT_DATA=true -e RP_IS_LOCAL_SERVER=false -e RP_USE_PERSONAL=false -e RP_USER=DNOCHEVN ******** quay.io/eclipse/che-e2e:next
```

**Comment by nallikaea** [comment]
no, you should not set RP_user by default, let me check

**Review by dmytro-ndp — APPROVED** [review]
looks good to merge

**Pull request merged**